universal/optimize_logos.sh

```
@@ -0,0 +1,211 @@
+            corner_brightness=$(echo "scale=2; ($nw_brightness + $ne_brightness + $sw_brightness + $se_brightness) / 4" | bc)
+            corner_brightness_int=$(printf "%.0f" "$corner_brightness")
+
+            # Determine background type: <=160 is dark, >160 is light
+            # This threshold works better for distinguishing dark grays from light backgrounds
+            if [ "$corner_brightness_int" -le 160 ]; then
+                bg_type="dark"
+                bg_color="Black"
+                target_color="black"
+                echo "  → Detected dark background (brightness: $corner_brightness_int)"
+            else
+                bg_type="light"
```

Shellcheck (0.11.0)

[warning] bg_type appears unused. Verify use (or export if used externally).

(SC2034)